framework/wazuh/core/active_response.py

```diff
@@ -13,84 +13,223 @@
+
+        Parameters
+        ----------
+        agent_version : str
+            The version of the agent.
+
+        Returns
+        -------
+        bool
+            True if ARStrMessage can handle the agent version, False otherwise.
+        """
+        return WazuhVersion(agent_version) < WazuhVersion(common.AR_LEGACY_VERSION)
+
+    def create_message(self, command: str = '', arguments: list = None, alert: dict = None) -> str:
+        """Create the message with the Active Response format that will be sent to the socket.
+
+        Parameters
+        ----------
+        command : str
+            Command running in the agent. If this value starts with !, then it refers to a script name instead of a command
+            name.
+        arguments : list
+            Command arguments.
+        alert : dict
+            Alert data that will be sent with the AR command.
+
+        Raises
+        ------
+        WazuhError(1650)
+            If the command is not specified.
+        WazuhError(1652)
+            If the command is not custom and the command is not one of the available commands.
+
+        Returns
+        -------
+        str
+            Message that will be sent to the socket.
+        """
+        if not command:
+            raise WazuhError(1650)
+
+        commands = get_commands()
+        if not command.startswith('!') and command not in commands:
+            raise WazuhError(1652)
+
+        msg_queue = command
+        msg_queue += " " + " ".join(shell_escape(str(x)) for x in arguments) if arguments else " - -"
+
+        return msg_queue
+
+
+class ARJsonMessage(ARMessageBuilder):
+    @staticmethod
+    def can_handle(agent_version: str) -> bool:
+        """Check if the ARJsonMessage can handle the given agent version.
+
+        Parameters
+        ----------
+        agent_version : str
+            The version of the agent.
+
+        Returns
+        -------
+        bool
+            True if ARJsonMessage can handle the agent version, False otherwise.
+        """
+        return WazuhVersion(agent_version) >= WazuhVersion(common.AR_LEGACY_VERSION)
+
+    def create_message(self, command: str = '', arguments: list = None, alert: dict = None) -> str:
+        """Create the message with the Active Response format that will be sent to the socket.
+
+        Parameters
+        ----------
+        command : str
+            Command running in the agent. If this value starts by !, then it refers to a script name instead of a command
+            name.
+        arguments : list
+            Command arguments.
+        alert : dict
+            Alert data that will be sent with the AR command.
+
+        Raises
+        ------
+        WazuhError(1650)
+            If the command is not specified.
+
+        Returns
+        -------
+        str
+            Message that will be sent to the socket.
+        """
+        if not command:
+            raise WazuhError(1650)
+
```

Comment:
Until now, it was only checked if a no-custom command really existed in the `ar.conf` for agents < 4.2.0, but not for the rest. 

I think this can be a bit tricky, an exception should be raised if a command is not custom and is not in `ar.conf` regardless of the agent version. Therefore, I would add all of this in a common place (`ARMessageBuilder.create_message`):

```PYTHON
        if not command:
            raise WazuhError(1650)

        commands = get_commands()
        if not command.startswith('!') and command not in commands:
            raise WazuhError(1652)
```

By the way, I would also update the exception `1652` to make the description more understandable. Something similar to:
```PYTHON
        1652: {'message': 'The command used is not defined in the configuration.',
               'remediation': f'Please, visit the official documentation (https://documentation.wazuh.com/'
                              f'{DOCU_VERSION}/user-manual/capabilities/active-response/how-to-configure.html)'
                              'to get more information'
               },
```

Reply:
Added validation:
https://github.com/wazuh/wazuh/blob/d335bbad035e6fe02ec2fc2adcd023cb27c60b99/framework/wazuh/core/active_response.py#L118-L140

And updated the `1652` description in commit d335bba.